### ISSUE TITLE
Meeting "GDPR - Purpose limitation" and the Broadcast Key

### ISSUE THREAD

**Issue by dirkx** [DP-3T/documents#155]
Right now - in the DP^3-T specification - there is a "Broadcast Key" defined. The exact value of this key determines the `realm' in which data can be shared. 

And most probably - one would want this key to be the same for all of Europe (for the COVID effort). As inter border traffic is common.

Meanwhile - as the member-states bring out their calls for tenders for actual implementations , the respondents have  show that DP^3-T meets one of the key principles of the GDPR - Purpose limitation (doelbinding).

As obviously an app like this has dual use / can live on beyond this Covid-19 event.

So once has to show in that this will not be the case. And that explicit, documented, technical measures (security/privacy by design/default) have been taken to do so.

So may I suggest that this key is set not to the value of "Broadcast Key" . But that  is set to "This broadcast key is only to be used in accordance to the EU Commission Guidelines C-2020-2296-final" (i.e. [1]). As the length of the string has no impact on protocl efficiency (it is hashed down to 32 bytes).

As this puts a technically tight `purpose limitation' on any actual implementation. And one that, though technically a `nonce' is administratively hard to subvert without a very clear, documented, decision trail.

I realise that this is a far cry from an academic paper -- where "Broadcast Key" is a perfect example. 

But ultimately this may get used in society - and hence it needs more than just cryptographic security.

1: https://ec.europa.eu/info/files/recommendation-apps-contact-tracing_en

**Comment by pdehaye** [comment]
In general, the choice of crypto functions and indeed the nonce are the tiniest bit of power left to democratic entities once Apple and Google implement this protocol. 

I find this proposal actually quite judicious in highlighting that Apple and Google are making (political) choices depriving health authorities of this power when backing apps.

**Comment by kennypaterson** [comment]
@dirkx I first would like to check that we have a common understanding that is only a fixed string used for the purpose of deriving keys, and not actually secret keying material.

**Comment by dirkx** [comment]
That is absolutely correct - it replaces the string which is now "Broadcast Key" in line https://github.com/DP-3T/reference_implementation/blob/3295ed7ddcf6c792d3e21cda19522a1302d8b015/LowCostDP3T.py#L29 and which is set to “Decentralized Privacy-Preserving Proximity Tracing” in some of the current other implementations.
